frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.tsx

```diff
@@ -200,7 +210,7 @@ export const ExpressionWidget = <Clause extends object = Lib.ExpressionClause>(
         <ExpressionEditorTextfield
           expression={expression}
           expressionIndex={expressionIndex}
-          clause={clause}
+          clause={clause as Lib.ExpressionClause | null}
```

Comment:
How do we know this type assertion is true?

AFAICT from the surounding code, clause can be `Clause | Lib.ExpressionClause | null`.

Maybe the type assertion on line 106 should be refined?

Reply:
> How do we know this type assertion is true?

Actually it isn't. It's a hack.

The thing is that `ExpressionEditorTextfield` component has been build with `Lib.ExpressionClause` in mind (and is typed with it). However it works perfectly fine with `Lib.AggregationClause` and `Lib.FilterableClause`. And we actually do use it for these types of clauses as well.

> Maybe the type assertion on line 106 should be refined?

No assertion in line 106. Which one are you referring to?

Anyway, I believe the fix is to make `ExpressionEditorTextfield` generic (and lots of utils from `/v1` dir), but the size of this change would be comparable to the size of this PR, hence this cast.

Reply:
I was referring to [line 106](https://github.com/metabase/metabase/pull/49949/files/76bef04f74ca37437e6bd5f22f4b20025df6a49f#diff-5588657ce99ef08f5ccc26a16c6587ac22653be47385aa3565c491edec11f504R106) in your PR. 

Your explanation makes sense, so don't let this block the PR.
Could you maybe add a TODO comment to explain why the cast is there?

Reply:
> I was referring to [line 106](https://github.com/metabase/metabase/pull/49949/files/76bef04f74ca37437e6bd5f22f4b20025df6a49f#diff-5588657ce99ef08f5ccc26a16c6587ac22653be47385aa3565c491edec11f504R106) in your PR, or [line 103] on `master`.

Unfortunately we don't have type guards for clause types (they would have to be implemented inside MLv2).

> Your explanation makes sense, so don't let this block the PR. Could you maybe add a TODO comment to explain why the cast is there?

Added: 4c72a1bd193df93ab4253225940b7bbe0b5e2fbe

Reply:
We might want to add `type Expressionable = ExpressionClause | FilterClause | AggregationClause;` in the lib and use it here

Reply:
> We might want to add `type Expressionable = ExpressionClause | FilterClause | AggregationClause;` in the lib and use it here

It's a larger change, I'll keep it out of this PR.